src/components/images/imageLoader.js

```diff
@@ -21,6 +21,37 @@ define(['lazyLoader', 'imageFetcher', 'layoutManager', 'browser', 'appSettings',
             return;
         }
 
+        try {
+            var blurHash = elem.getAttribute("data-blurhash");
+            if (blurHash && blurhash.isBlurhashValid(blurHash)) {
+                var width = 85; // TODO: get correct dimensions
+                var height = 128;
+
+                var pixels = blurhash.decode(blurHash, width, height);
+
+                if (!self.canvas || !self.ctx) {
+                    var canvas = document.createElement('canvas');
+                    canvas.width = width;
+                    canvas.height = height;
+
+                    self.canvas = canvas;
+                    self.ctx = canvas.getContext('2d');
+                }
+
+                var ctx = self.ctx;
+                var imageData = ctx.createImageData(width, height);
+                imageData.data.set(pixels);
+                ctx.putImageData(imageData, 0, 0);
+                var uri = self.canvas.toDataURL();
+                elem.style.backgroundImage = "url('" + uri + "')";
+                enableEffects = false;
+            }
+        } catch (e) {
+            console.log("blurhash load failed: " + e.toString());
+        }
```

Comment:
Which commands require a try-catch block?

Reply:
Just `blurhash.decode` as far as I know; I'll double check and make this clearer in the code.